tests/test_reduce_api_funcs.py

```diff
@@ -0,0 +1,215 @@
+import numpy as np
+from numpy.testing import assert_equal
+import pytest
+import finch
+
+# verbose = False
+verbose = True
+
+def output_term(args):
+    if verbose:
+        print(args)
+
+
+# TODO: (Overall) 3-D or higher dimension tensor testcases
+
+@pytest.mark.parametrize(   
+    "a",
+    [
+        (np.array([[2, 3, 4], [5, 6, 7]])),
+    ],
+)
+def test_lazyTensor_prod(a):
+    output_term("")
+    output_term("***** prod *****")
+
+    # TODO: axis=None is not supported well in all interfaces?
+    # TODO: results computed from operator.mul are all 0?
+
+    # result = finch.compute(finch.prod(finch.lazy(a), axis=None))
+    # output_term(result)
+    # expected = np.prod(a)
+    # output_term(expected)
+    # assert_equal(result, expected)
+    # output_term("")
+
+    result = finch.compute(finch.prod(finch.lazy(a), axis=0)); output_term(result)
+    expected = np.prod(a, axis=0); output_term(expected)
+    # assert_equal(result, expected)
+    output_term("")
+
+    result = finch.compute(finch.prod(finch.lazy(a), axis=1)); output_term(result)
+    expected = np.prod(a, axis=1); output_term(expected)
+    # assert_equal(result, expected)
+    output_term("")
+
+
+@pytest.mark.parametrize(
```

Comment:
IN general, you can probably group tests into three groups: float arrays, int arrays, and bool arrays, where each group tests all applicable methods for each.

Reply:
Group the tests into three groups: bitwise reductions, integer reductions, and float reductions. Use `@parameterize` to iterate over a list of the applicable ops for each group.